internal/controllers/operator_controller.go

```diff
@@ -264,18 +271,28 @@ func (r *OperatorReconciler) bundleFromSolution(solution *solver.Solution, packa
 	return nil, fmt.Errorf("bundle for package %q not found in solution", packageName)
 }
 
-func (r *OperatorReconciler) generateExpectedBundleDeployment(o operatorsv1alpha1.Operator, bundlePath string, bundleProvisioner string) *unstructured.Unstructured {
+func (r *OperatorReconciler) generateExpectedBundleDeployment(o operatorsv1alpha1.Operator, bundle *catalogmetadata.Bundle, bundleProvisioner string) (*unstructured.Unstructured, error) {
 	// We use unstructured here to avoid problems of serializing default values when sending patches to the apiserver.
 	// If you use a typed object, any default values from that struct get serialized into the JSON patch, which could
 	// cause unrelated fields to be patched back to the default value even though that isn't the intention. Using an
 	// unstructured ensures that the patch contains only what is specified. Using unstructured like this is basically
 	// identical to "kubectl apply -f"
 
+	bundleVersion, err := bundle.Version()
+	if err != nil {
+		return nil, err
+	}
+
 	bd := &unstructured.Unstructured{Object: map[string]interface{}{
 		"apiVersion": rukpakv1alpha1.GroupVersion.String(),
 		"kind":       rukpakv1alpha1.BundleDeploymentKind,
 		"metadata": map[string]interface{}{
 			"name": o.GetName(),
+			"annotations": map[string]string{
+				variablesources.AnnotationPackageName:   bundle.Package,
+				variablesources.AnnotationBundleName:    bundle.Name,
+				variablesources.AnnotationBundleVersion: bundleVersion.String(),
+			},
```

Comment:
Should we make these labels (rather than annotations) so that we can use a label selector to populate our BD cache with _just_ BDs that we are managing?

Reply:
@joelanford I think we already do that but in a different way.

We set owner reference on a BD here:

https://github.com/operator-framework/operator-controller/blob/205a48778e07f96819381269cb652536c6f895e6/internal/controllers/operator_controller.go#L297-L306

And only reconcile owned objects:

https://github.com/operator-framework/operator-controller/blob/205a48778e07f96819381269cb652536c6f895e6/internal/controllers/operator_controller.go#L316

Reply:
Kubernetes does not support setting up informers to filter on the server-side using owner references though. So my suggestion is to use labels so that we can _also_ configure the cache to select only BDs that have the expected label set.